smalltalksrc/Slang-Tests/SlangBasicTranslationTest.class.st

```diff
@@ -1155,6 +1155,163 @@ SlangBasicTranslationTest >> testGoto [
 	self assert: translation equals: 'goto lab'
 ]
 
+{ #category : #'tests-inline-builtins' }
+SlangBasicTranslationTest >> testInlineMethodSumArgumentsNoPrepare [
+	
+	| tMethod translation |
+	tMethod := (self getTMethodFrom: #methodSumParameters:to:with:).
+	generator doBasicInlining: true.
+	
+	translation := self translate: tMethod.
+	translation := translation trimBoth.
+	self
+		assert: translation
+		equals: '/* SlangBasicTranslationTestClass>>#methodSumParameters:to:with: */
+static sqInt
+methodSumParameterstowith(sqInt pFrom, sqInt pTo, sqInt anInteger)
+{
+	sqInt i;
+	sqInt pFrom1;
+	sqInt pTo1;
+
+	/* begin methodFrom:to: */
+	pFrom1 = pFrom + anInteger;
+	pTo1 = pTo + anInteger;
+	for (i = 0; i <= 5; i += 1) {
+		pTo1[i] = pFrom1[i];
+	};
+	return 0;
+}'
+]
+
+{ #category : #'tests-inline-builtins' }
+SlangBasicTranslationTest >> testInlineMethodSumArgumentsPrepared [
+	
+	| tMethod translation |
+	tMethod := (self getTMethodFrom: #methodSumParameters:to:with:).
+	generator 
+		prepareMethods;
```

Comment:
I think preparing methods should be mandatory, as Slang does it all the time when generating the VM.
I don't think Slang is prepared to translate something if this is not done.

Actually, what we would like to test is:
 - a test where the operand is inlined in the array access `(a + b)[i] = ...` => this we have it
 - a test where the array pointer is outlined into a variable `var := expression; expression[i] = ...` => this we can do it by having a piece of code that cannot be inlined. E.g., a function call!

Reply:
Ok. I've updated the PR with the suggested changes.